src/advection/bsl_advection_polar.hpp

```diff
@@ -203,6 +218,13 @@ class BslAdvectionPolar
         IdxRangeR radial_grid(grid);
         IdxRangeBatchedWithoutR no_r_grid(grid);
 
+        // Check the first points on R correspond to the O-point.
+        assert(norm_inf(
+                       m_logical_to_physical_mapping(ddc::coordinate(
+                               Idx<GridR, GridTheta>(radial_grid.front(), no_r_grid.front())))
+                       - m_logical_to_physical_mapping.o_point())
+               < 1e-13);
```

Comment:
Why do you need to be in Cartesian coordinates to check if the first points on R correspond to the O-point? Is it not sufficient to do:
```suggestion
        // Check the first points on R correspond to the O-point.
        assert(ddc::coordinate(radial_grid.front()) < 1e-13);
```
?

Reply:
It was also to cover weird geometries for those the Opoint does not correspond to r = 0.  As we have already noticed that the O-point isnot at (x,y) = (0,0)

Reply:
Can you give me an example of a weird geometry that you would like to support?
Even if the O-point is at x=42 it is still at r=0